### PR TITLE
Branch view details

### LINKED ISSUE #50: Update view command to see other data
As an Editor, I want to view the data of the authors under me: 
1. As an Editor, I want to view all the authors under me so that I can keep track of them. 
2. As an Editor, I want to view all the manga under a specific author so that I can keep track of them. 
3. As an Editor, I want to view deadlines set for each manga under a specific author so that I can monitor their progress. 
4. As an Editor, I want to view sales data (unit price and quantity sold) for each manga under a specific author so that I can monitor the company's finances. 
>Note: the first 2 have already been done, but require updates and are included here for completeness. 

Additional todo:
- [ ] Write JUnit tests for existing view authors and view mangas commands

### PR DESCRIPTION
**Pull request by xenthm**
Fixes #50

### CODE REVIEW

**Comment by xenthm** [comment]
`view` sample output:
image

`view -a test1` samples output:
image

File: src/main/java/parser/Parser.java
Comment (xenthm) [inline src/main/java/parser/Parser.java:1]: @averageandyyy do take a look at this new parser logic. We can look into expanding this for the other commands.

Comment (xenthm) [reply src/main/java/parser/Parser.java:1]: BTW the flags can be in any order!

**Comment by xenthm** [comment]
This implementation allows for incremental inclusion of `Manga` attributes to be printed. 

`view -a test1 -d` sample output:
image

`view -a test1 -s` sample output:
image


`view -a test1 -d -s` sample output:
image

File: src/main/java/ui/PrintColumn.java
Comment (sarahchow03) [inline src/main/java/ui/PrintColumn.java:1]: I like how this displays all the information in a organized manner. Will be useful for the user when a lot titles are logged into Tantou.

File: src/main/java/constants/Regex.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 public class Regex {
     public static final String SPACE_REGEX = " ";
     public static final String EMPTY_REGEX = "";
+    public static final String DIVIDER_REGEX = " | ";
```
Comment (sarahchow03) [inline src/main/java/constants/Regex.java:14]: Good practice of labelling the divider for clarity.

File: src/main/java/manga/MangaList.java [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
+        if (includeSales) {
+            columns.add(new PrintColumn<>("Unit Price", 10, manga -> {
+                Sale salesData = manga.getSalesData();
+                return salesData != null ? String.valueOf(salesData.getUnitPrice()) : "N/A";
+            }));
+            columns.add(new PrintColumn<>("Units Sold", 10, manga -> {
+                Sale salesData = manga.getSalesData();
+                return salesData != null ? String.valueOf(salesData.getQuantitySold()) : "N/A";
+            }));
+            columns.add(new PrintColumn<>("Revenue", 10, manga -> {
```
Comment (sarahchow03) [inline src/main/java/manga/MangaList.java:32]: What do the magic numbers `40`, `3`, `15` in these lines mean?

**Review by sarahchow03 — APPROVED** [review]
LGTM!

File: src/main/java/manga/MangaList.java (hunk 2 above)
Comment (xenthm) [reply src/main/java/manga/MangaList.java:32]: They are pre-set widths for the columns.

**Pull request merged**